### ISSUE TITLE
Resetting stub server

### ISSUE THREAD

**Issue by 0x6e6562** [mkotsur/restito#29]
I was wondering if there is a way to reset the state of a stub server. My use case is when I want to verify that a particular URL is only ever invoked once. My app is making a call to a remote HTTP URL (which I am stubbing out with restito), but it protects against duplicate invocations of the downstream HTTP server. So my test invokes my application twice, but only expects the stub to invoked once. What I think I'm looking for is a call to reset the state of the stub server so that I can then assert 0 invocations, after first having asserted 1 invocation.

**Comment by mkotsur** [comment]
Sorry, I didn't understand the use case completely. If you want to assert that the call happens only once, this is exactly what restito tells: "No, it happens 2 times.". Probably I am missing some more context about "protects against duplicate invocations of the downstream HTTP server".

But in any case, it's possible to reset the state simply by calling `server.getCalls().clear()`. Feel free to reopen the issue if it doesn't solve your problem.

**Comment by 0x6e6562** [comment]
```
server.getCalls().clear()
```

solved my issue, so thank you very much for this.

For anybody googling this, the use case is a client calling a server which in turn calls an SMS gateway to deliver a message. If the client submits a duplicate request to the server, the server should not send a request to the SMS gateway. So there are 2 client calls, but the second is a duplicate of the first:

1. Client -> Real Server -> Stubbed out SMS gateway
2. Client -> Real Server -> Call to SMS gateway should not occur

In this scenario I want to verify exactly one verification that the stub was invoked after the first client call but before the second. And then after the second I want to assert that stub was not invoked.

I could just run both client calls through and then assert after them both that the stub was only invoked once in total. This has some value and is simpler. But in this particular case I need the more fine grained verification that the first client call causes the stub to get invoked and not the second, rather than just asserting the any one of the client calls cause the stub to be invoked.

**Comment by mkotsur** [comment]
Aha, now it's clear. 

Assuming that you test the `Real Server`, you stubbed the `SMS gateway` and you have access to the client in the test, I'd write the test like that:

```java
whenHttp(server).doAllResponsesSetup();
client.doCall();
verifyHttp(server).once();
client.doCall();
verifyHttp(server).once(); // Will ensure that even after the second client action, the server has been called only once.
```

**Comment by 0x6e6562** [comment]
Many thanks for this suggestion - I've now integrated this into my test suite and it makes things a lot clearer :-)
